mitiq/calibration/calibration.py

```diff
@@ -0,0 +1,224 @@
+                )
+                improvement_factor = abs(noisy_value - ideal_value) / sqrt(
+                    len(mitigated_values)
+                    * sum(
+                        (mitigated_value - ideal_value) ** 2
+                        for mitigated_value in mitigated_values
+                    )
+                )
+                di["improvement_factor"] = improvement_factor
+
+    def best_strategy(self, results: List[Dict[str, Any]]) -> Strategy:
+        """Finds the best strategy by using the parameters that had the
+        largest improvement factor."""
+        best_improvement_factor = 0.0
+        strategy = None
+        for circuit in results:
+            for result in circuit["mitigated_values"]["zne"]["results"]:
+                if result["improvement_factor"] > best_improvement_factor:
+                    best_improvement_factor = result["improvement_factor"]
+                    strategy = result["strategy"]
+        if strategy is None:
+            raise ValueError("None of the improvement factors were > 0")
+        return strategy
+
+    def run(self) -> None:
+        results = self.run_circuits()
+        self.compute_improvements(results)
+        self.results = results
+
+
+def bitstrings_to_distribution(
+    bitstrings: List[List[int]],
+) -> Dict[str, float]:
+    """Helper function to convert raw measurement results to probability
+    distributions."""
+    distribution = Counter(
+        ["".join(map(str, bitstring)) for bitstring in bitstrings]
+    )
+    bitstring_count = len(bitstrings)
+    for bitstring in distribution:
+        distribution[bitstring] /= bitstring_count  # type: ignore [assignment]
+    return dict(distribution)
+
+
+def convert_to_expval_executor(
+    ex: Executor,
+    distribution: Optional[Dict[str, float]] = None,
+    bitstring: Optional[str] = None,
+) -> Tuple[Executor, str]:
+    """Constructs a new executor returning an expectation value given by the
+    probability that the circuit outputs the most likely state according to the
+    ideal distribution.
+
+    Returns:
+        A two-tuple containing:
+            - An executor returning expectation values: QPROGRAM -> float
+            - The most likely bitstring, according to the passed `distribution`
+    """
+    bitstring_to_measure = ""
+    if distribution:
+        bitstring_to_measure = max(
+            distribution,
+            key=distribution.get,  # type: ignore [arg-type]
+        )
+    elif bitstring:
+        bitstring = bitstring
+
+    def expval_executor(circuit: cirq.Circuit) -> float:
+        raw = cast(MeasurementResult, ex._run([circuit])[0]).result
+        bitstring_distribution = bitstrings_to_distribution(raw)
+        return bitstring_distribution.get(bitstring_to_measure, 0)
+
+    return (
+        Executor(expval_executor),  # type: ignore [arg-type]
+        bitstring_to_measure,
+    )
+
+
+def execute_with_mitigation(
+    circuit: QPROGRAM,
+    calibrator: Calibrator,
```

Comment:
We could use the same signature of all the Mitiq execute_with_xxx functions:

```suggestion
    circuit: QPROGRAM,
    executor: Union[Executor, Callable],
    observable: Optional[Observable] = None,
    *,
    calibrator: Calibrator,
```